web/src/DetailedInfoPage.tsx

```diff
@@ -0,0 +1,499 @@
+/* eslint react-hooks/exhaustive-deps: 0 */
+
+import * as React from "react";
+
+import { useParams, useNavigate } from "react-router-dom";
+import {
+    FamilyApi,
+    SampleApi,
+    ParticipantApi,
+    SequenceApi,
+    ParticipantModel,
+    Sample,
+    SampleSequencing,
+} from "./sm-api/api";
+import { Table } from "semantic-ui-react";
+
+// TODO Move interfaces to appropriate API/routes
+interface PedigreeEntry {
+    affected: number;
+    family_id: string;
+    individual_id: string;
+    maternal_id: string;
+    paternal_id: string;
+    sex: number;
+}
+interface MetadataReads {
+    location: string;
+    basename: string;
+    class: string;
+    checksum: string;
+    size: number;
+}
+
+const sampleFieldsToDisplay = ["active", "type", "participant_id"];
+
+export const DetailedInfoPage: React.FunctionComponent<{}> = () => {
+    const { projectName, sampleName } = useParams();
+    const navigate = useNavigate();
+    const [samples, setSamples] = React.useState();
+    const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [error, setError] = React.useState<string | undefined>();
+    const [sample, setSample] = React.useState<string>();
+    const [sampleInfo, setSampleInfo] = React.useState<Sample>();
+    const [pedigree, setPedigree] = React.useState<PedigreeEntry[]>();
+    const [participants, setParticipants] =
+        React.useState<ParticipantModel[]>();
+    const [selectedExternalID, setSelectedExternalID] =
+        React.useState<string>();
+    const [idNameMap, setIdNameMap] = React.useState<[string, string][]>();
+    const [sequenceInfo, setSequenceInfo] = React.useState<SampleSequencing>();
+
+    const getSamplesFromProject = React.useCallback(async () => {
```

Comment:
I think this is fine for first iteration, but we should move all this logic to the web layer, and have a "get family display info for web" sort of thing.

Reply:
Yeah definitely, would be great to have fewer (or 1) API call which gets the all relevant info